File: align_parser/src/align_parsing/Segment.java
package align_parsing;

import java.util.ArrayList;
import java.util.List;

public class Segment {
    int startWordIndex;
    int endWordIndex;
    int startMillis;
    int endMillis;

    public int getEndMillis() {
        return endMillis;
    }

    public int getEndWordIndex() {
        return endWordIndex;
    }

    public int getStartWordIndex() {
        return startWordIndex;
    }

    public int getStartMillis() {
        return startMillis;
    }

    public static List<Segment> makeSegmentsFromIntArray(int[][] segArr) {
        ArrayList<Segment> segments = new ArrayList<>();

        for (int[] segment : segArr) {
            Segment s = new Segment();
            s.startWordIndex = segment[0];
            s.endWordIndex = segment[1];
            s.startMillis = segment[2];
            s.endMillis = segment[3];
            segments.add(s);
        }

        return segments;
    }
}

File: align_parser/src/align_parsing/Segment.kt
package align_parsing

import java.util.*

internal class Segment(
    val startWordIndex: Int, val endWordIndex: Int,
    val startMillis: Int, val endMillis: Int
) {
    private var text: String = ""
    fun setText(text: String) {
        this.text = text
    }

    fun getText(): String {
        return text
    }

    companion object {

        fun makeSegmentsFromIntArray(segArr: Array<IntArray>): List<Segment> {
            val segments = ArrayList<Segment>()

            for (segment in segArr) {
                val startWordIndex = segment[0]
                val endWordIndex = segment[1]
                val startMillis = segment[2]
                val endMillis = segment[3]
                segments.add(Segment(startWordIndex, endWordIndex, startMillis, endMillis))
            }

            return segments
        }
    }

    override fun toString(): String {
        return "[$startWordIndex:$endWordIndex]\t$text"
    }
}
